notifications/src/main/java/org/openedx/notifications/presentation/primer/NotificationsPrimerDialogFragment.kt

```diff
@@ -113,10 +111,43 @@ class NotificationsPrimerDialogFragment : DialogFragment() {
                             CircularProgressIndicator(color = MaterialTheme.appColors.primary)
                         }
                     }
+
+                    PrimerUIState.ShowRationalDialog -> {
+                        ShowRationalePermissionDialog(
+                            onNegativeButtonClick = {
+                                viewModel.dismissDialog()
+                            },
+
+                            onPositiveButtonClick = {
+                                PermissionUtils.navigateToNotificationSettings(
+                                    requireContext()
+                                )
+                                viewModel.dismissDialog()
+                            },
+                        )
+                    }
                 }
             }
         }
     }
+
+    @Composable
+    private fun ShowRationalePermissionDialog(
```

Comment:
Please move it before preview as the main UI composables should be first in the file.